src/backend/controllers/userController.js

```diff
@@ -0,0 +1,169 @@
+                email,
+                password
+            });
+
+            // Creates salt with 10 rounds(recommended)
+            const salt = await bcrypt.genSalt(10);
+
+            // bcyrpt hash passwords
+            user.password = await bcrypt.hash(password, salt);
+
+            // Save to MongoDB
+            await user.save();
+
+            const payload = {
+                user: {
+                    // Primary key id generated by Mongo
+                    id: user.id
+                }
+            }
+
+            // Need to manage the config path so that the jwttoken is not here see require('config')
+            // expiresIn 4 hrs
+            jwt.sign(payload, process.env.SECRET, 
+            { expiresIn: 14400},
+            (err, token) => {
+                if(err != null) throw err;
+                res.json({ token });
+            });
+        } catch(err) {
+            console.error(err.message);
+            res.status(500).json({ msg: 'Server Error'});
+        }
+}
+
+exports.login = async (req: $Request, res: $Response) => {
+    // Makes sure the signup is valid
+    const errors = validationResult(req);
+    if(!errors.isEmpty()){
+        // 400 is a bad request
+        return res
+            .status(400)
+            .json({ errors: errors.array() });
+    }
+
+    // Assigns request body to user schema fields
+    const { userName, password } = req.body;
+
+    try{
+        // See if User exists - might change this to const
+        let user = await User.findOne({ userName });
+        if(user == null){
+            return res
+                .status(400)
+                .json({ errors: [ {msg: 'Invalid Credentials' }] });
+        }
+
+        // Checks if the plaintext password matches the hashed pass form db
+        const isMatch = await bcrypt.compare(password, user.password);
+
+        // If the credentials dont match it will return a boolean false
+        if(!isMatch) {
+            return res
+                .status(400)
+                .json({ errors: [ { msg: 'Invalid Credentials' }] });
+        }
+
+
+        const payload = {
+            user: {
+                // Primary key id generated by Mongo
+                id: user.id
+            }
+        }
+
+        // expiresIn 4 hour 
+        jwt.sign(payload, process.env.SECRET, 
+        { expiresIn: 14400},
+        (err, token) => {
+            if(err) throw err;
+            res.json({ token });
+        });
+    } catch(err) {
+        console.error(err.message);
+        res.status(500).json({ msg:'Server error'});
+    }
+
+}
+
+
+exports.getUser = async ( req: $Request, res: $Response) => {
+        try{
+            const user = await User.findById(req.params.userId);
+            res.json(user);
+        } catch(err){
+            console.error(err.message);
+            res.status(500).json({ msg: 'Server Error'});
+        }
+}
+
+/*exports.updateCassus = async (req: $Request, res: $Response){
```

Comment:
Delete this if it isn't working and then publish it in a later push if you need to